analysis/type.go

```diff
@@ -82,6 +86,11 @@ type DefaultAnalyzer struct {
 	TokenFilters []TokenFilter
 }
 
+type WrapperAnalyzer struct {
```

Comment:
I'd suggest making this a bit more generic. for example the AdditionalTokenFilter field can be collection of TokenFilters, that way any non-synonym use case may use this.